starter/source/output/th/hm_read_thgrpa.F

```diff
@@ -82,7 +82,64 @@ SUBROUTINE HM_READ_THGRPA(
       ITYP = 1001     ! type /th/part
 
       ! read option header
-      CALL HM_OPTION_READ_KEY(LSUBMODEL, OPTION_ID=TH_ID, OPTION_TITR=TITR, KEYWORD2=KEY)   
+      CALL HM_OPTION_READ_KEY(LSUBMODEL, OPTION_ID=TH_ID, OPTION_TITR=TITR, KEYWORD2=KEY, KEYWORD3=FLAG_ALL)   
+
+      IF (FLAG_ALL(1:3) == 'ALL') THEN !The new keyword for part-time-history: /TH/PART/ALL/...  
+
+      ! Number of variables counted in input line
+      CALL HM_GET_INTV('Number_Of_Variables',NVAR,IS_AVAILABLE,LSUBMODEL) 
```

Comment:
@FanshiFabienMENG 

could you reindent all this part please ?

Thanks,
Sébastien

Reply:
Hello @sebastienVilleneuve ,

Yes, I will do it. And we still have some discussions about this modification. Let me put it in dratf.

Thanks;
Fanshi